pkg/controller/core.oam.dev/v1alpha2/application/application_controller.go

```diff
@@ -349,6 +373,15 @@ func appWillRollout(app *v1beta1.Application) bool {
 	return len(app.GetAnnotations()[oam.AnnotationAppRollout]) != 0 || app.Spec.RolloutPlan != nil
 }
 
+func hasHealthCheckPolicy(policies []*unstructured.Unstructured) bool {
```

Comment:
you should check policyDefinition for this mechanism. please add a field named `manageHealthCheck` into PolicyDefinition